Controllers/ProductsController.cs

```diff
@@ -44,6 +56,70 @@ public ProductsController(BangazonContext ctx)
             return View(product);
         }
 
+
+        public async Task<IActionResult> EditInfo([FromRoute]int? id)
+        {
+            ViewData["ProductTypeId"] = context.ProductType
+                .OrderBy(l => l.Label)
+                .AsEnumerable()
+                .Select(li => new SelectListItem { 
+                    Text = li.Label,
+                    Value = li.ProductTypeId.ToString()
+                    });
+                                        
+            // If no id was in the route, return 404
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var product = await context.Product
+                    .Include(s => s.User)
+                    .SingleOrDefaultAsync(m => m.ProductId == id);
+
+            // If product not found, return 404
+            if (product == null)
+            {
+                return NotFound();
+            }
+
+            return View(product);
+        }
+
+        [ValidateAntiForgeryToken]
+        public async  Task <IActionResult> Edit([FromRoute]int id, Product product)

```

Comment:
We should make id nullable to get rid of the red squiggly on line 97.
`public async  Task <IActionResult> Edit([FromRoute]int? id, Product product)`